src/creature.h

```diff
@@ -242,6 +242,12 @@ class Creature
         virtual const avatar *as_avatar() const {
             return nullptr;
         }
+        virtual monster *as_monster() {
```

Comment:
You need to add overrides for those two functions to the `monster` class. Currently `some_monster.as_monster()` will call this function and therefor always return 0. See implementation of `as_player` in class `player`.